Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py

```diff
@@ -260,50 +271,100 @@ def main() -> None:
-            cases_url = '/cases/' + demisto.args().get('status')
+            status = arguments.get('status')
+            cases_url = '/cases/' + status
             fetch_records(client, cases_url, 'Gra.Cases', 'caseId', params)
 
         elif demisto.command() == 'gra-user-anomalies':
-            anomaly_url = '/users/' + demisto.args().get('employee_id') + '/anomalies/'
+            employee_id = arguments.get('employee_id')
+            anomaly_url = '/users/' + employee_id + '/anomalies/'
             fetch_records(client, anomaly_url, 'Gra.User.Anomalies', 'anomaly_name', params)
 
+        elif demisto.command() == 'gra-case-action':
+            action = arguments.get('action')
+            caseId = arguments.get('caseId')
+            subOption = arguments.get('subOption')
+            caseComment = arguments.get('caseComment')
+            riskAcceptDate = arguments.get('riskAcceptDate')
+            cases_url = '/cases/' + action
+            if action == 'riskManageCase':
+                post_url = '{"caseId":' + caseId + ',"subOption":"' + subOption + '","caseComment":"' + caseComment +\
+                           '","riskAcceptDate":"' + riskAcceptDate + '"}'
+            else:
+                post_url = '{"caseId":' + caseId + ',"subOption":"' + subOption + '","caseComment":"' + \
+                           caseComment + '"}'
+            fetch_post_records(client, cases_url, 'Gra.Case.Action', 'caseId', params, post_url)
+
+        elif demisto.command() == 'gra-case-action-anomaly':
+            action = arguments.get('action')
+            caseId = arguments.get('caseId')
+            anomalyNames = arguments.get('anomalyNames')
+            subOption = arguments.get('subOption')
+            caseComment = arguments.get('caseComment')
+            riskAcceptDate = arguments.get('riskAcceptDate')
+            cases_url = '/cases/' + action
+            if action == 'riskAcceptCaseAnomaly':
+                post_url = '{"caseId":' + caseId + ',"anomalyNames":' + anomalyNames + ',"subOption":"' + subOption +\
+                           '","caseComment":"' + caseComment + '","riskAcceptDate":"' + riskAcceptDate + '"}'
+            else:
+                post_url = '{"caseId":' + caseId + ',"anomalyNames":"' + anomalyNames + '","subOption":"' + \
+                           subOption + '","caseComment":"' + caseComment + '"}'
+            fetch_post_records(client, cases_url, 'Gra.Cases.Action.Anomaly', 'caseId', params, post_url)
+
+        elif demisto.command() == 'gra-investigate-anomaly-summary':
+            fromDate = arguments.get('fromDate')
+            toDate = arguments.get('toDate')
+            modelName = arguments.get('modelName')
+            if fromDate is not None and toDate is not None:
+                investigateAnomaly_url = '/investigateAnomaly/anomalySummary/' + modelName + '?fromDate=' + fromDate \
+                                         + ' 00:00:00&toDate=' + toDate + ' 23:59:59'
```

Comment:
Should the query parameters get both a value from the user and the predefined values of `00:00:00` and `23:59:59`?

Also, you can add the `fromDate` and `toDate` query parameters to the `params` object that was defined in #L231.
```suggestion
                investigateAnomaly_url = '/investigateAnomaly/anomalySummary/' + modelName
                params['fromDate'] = f'{fromDate} 00:00:00'
                params['toDate'] =  f'{toDate} 23:59:59'
                
```